Examples/test-suite/python/python_overload_simple_cast_runme.py

```diff
@@ -164,34 +164,120 @@ def __float__(self):
     raise RuntimeError("Spam(void *)")
 
 
+# nextafter: ++ and -- operators for float, off by one LSB
+# nextafter was released in Python 3.9
+from sys import hexversion
+from math import inf
+if hexversion >= 0x3090000:
+    from math import nextafter
+else:
+    # Else try numpy as they provide this functions for decades
+    import numpy
+    def nextafter(x,y):
+        return float(numpy.nextafter(x, y))
+
+def exceptTypeError(fun, arg, msg):
+    try:
+        fun(arg)
+        raise RuntimeError(msg)
+    except TypeError:
+        pass
+
+# x86_64: long is 32bit on MSVC but 64bit on *nix
+assert sizeof_long() in [4, 8]
+
+# unsigned long
+ulmax = 2**32 - 1
+ulmin = 0
+ulmaxd = float(2**32-1)
+ulmind = 0.0
+if sizeof_long() == 8:
+    ulmax = 2**64 - 1
+    ulmaxd = nextafter(float(2**64), 0)
+
+if as_ul(ulmin) != ulmin:
+    raise RuntimeError("as_ul(ulmin)")
+if as_ul(ulmax) != ulmax:
+    raise RuntimeError("as_ul(ulmax)")
+if as_ul(ulmind) != ulmind:
+    raise RuntimeError("as_ul(ulmind)")
+if as_ul(ulmaxd) != ulmaxd:
+    raise RuntimeError("as_ul(ulmaxd)")
+
+exceptTypeError(as_ul, ulmin - 1, "as_ul(ulmin - 1)")
+exceptTypeError(as_ul, ulmax + 1, "as_ul(ulmax + 1)")
+exceptTypeError(as_ul, nextafter(ulmind, -inf), "as_ul(ulmind - LSB)")
+exceptTypeError(as_ul, nextafter(ulmind, inf), "as_ul(ulmaxd + LSB)")
+
+# long
+lmax = 2**31 - 1
+lmin = -2**31
+lmaxd = float(2**31 - 1)
+lmind = float(-2**31)
+if sizeof_long() == 8:
+    lmax = 2**63 - 1
+    lmin = -2**63
+    lmaxd = nextafter(float(2**63), 0)
+    lmind = float(-2**63)
+
+if as_l(lmin) != lmin:
+    raise RuntimeError("as_l(lmin)")
+if as_l(lmax) != lmax:
+    raise RuntimeError("as_l(lmax)")
+if as_l(lmind) != lmind:
+    raise RuntimeError("as_l(lmind)")
+if as_l(lmaxd) != lmaxd:
+    raise RuntimeError("as_l(lmaxd)")
+
+exceptTypeError(as_l, lmin - 1, "as_l(lmin - 1)")
+exceptTypeError(as_l, lmax + 1, "as_l(lmax + 1)")
+exceptTypeError(as_l, nextafter(lmind, -inf), "as_l(lmind - LSB)")
+exceptTypeError(as_l, nextafter(lmaxd, inf), "as_l(lmaxd + LSB)")
+
 # unsigned long long
-ullmax = 9223372036854775807  # 0xffffffffffffffff
```

Comment:
Note: This does NOT match. The old value `9223372036854775807 == 2**63-1` conflicts with the comment.

Reply:
Agreed, `9223372036854775807 == 0x7fffffffffffffff`